src/domain-object.ts

```diff
@@ -6,15 +6,15 @@ export abstract class DomainObject<TData extends object = {}> extends Serializab
     /**
      * @param value (the value to compare)
      */
-    public equals(value: DomainObject | null): boolean
+    public equals(value?: DomainObject | null): boolean
```

Comment:
Because this will make it  `DomainObject | null | undefined`. Sometimes we look for things in an array for eg: with `find` the return can be `undefined`. In this case, we need to explicitly check for `== null` first, before using the equals.

Reply:
Why can't we use the equals right away?

Reply:
This is making `.equals` more excepting

Reply:
This by interface design allows you to call equals with nothing. Fact that the array can return undefined is a javascript quirk. No other language does that. We shouldn't define bad interfaces based on js quirks. Because now I can have code like this: ```if(foo.equals())``` // like this looks stupid. If you want to support your array scenario, then what you want is ```public equals(value: DomainObject | null | undefined): boolean```. There is a difference between allowing undefined as an input vs declaring something as an optional parameter.